ksql-engine/src/main/java/io/confluent/ksql/metrics/Counter.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.ksql.metrics;
+
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+class Counter<R> {
+
+  private final String topic;
+  Map<String, SensorMetric<R>> sensors = new HashMap<>();
```

Comment:
Can we make this `private final` and add methods to the class like `findFirst` etc so that `ProducerCollector` and `ConsumerCollector` don't need access to the `sensors` map?

Reply:
ok - done